app/src/main/java/com/example/userinfo/MainActivity.kt

```diff
@@ -0,0 +1,182 @@
+            return false
+        }
+        if (!isValidPhone(number.text.toString())) {
+            callToast("Invalid Phone Number")
+            return false
+        }
+        if(!isValidUserName(userName.text.toString())){
+            callToast("Invalid User Name")
+            return false
+        }
+        return true
+    }
+
+    private fun callToast(str:String){
+        Toast.makeText(applicationContext, str,
+            Toast.LENGTH_SHORT).show()
+    }
+
+    private fun checkAllFields(
+        userName: EditText,
+        email: EditText,
+        number: EditText,
+        pincode: EditText,
+        address: EditText
+    ): Boolean {
+
+        if (userName.text.trim().isEmpty()) {
+            Log.i("UserName", userName.id.toString())
+            callToast("Username shouldn't be empty")
+            return false
+        }
+        if (email.text.trim().isEmpty()) {
+            callToast("Email shouldn't empty")
+            return false
+        }
+        if(number.text.trim().isEmpty()) {
+            callToast("Phone number shouldn't be empty")
+            return false
+        }
+        if(pincode.text.trim().isEmpty()){
+            callToast("Pincode shouldn't be empty")
+            return false
+        }
+        if(address.text.trim().isEmpty()){
+            callToast("Address shouldn't be empty")
+            return false
+        }
+        return true
+    }
+    fun isValidEmail(str: String): Boolean {
+        return emailPattern.matcher(str).matches()
+    }
+
+    fun isValidPinCode(str: String): Boolean {
+        return pincodePattern.matcher(str).matches()
+    }
+
+    fun isValidPhone(str: String): Boolean {
+        return phoneNumberPattern.matcher(str).matches()
+    }
+
+    fun isValidUserName(str:String):Boolean{
+        return userNamePattern.matcher(str).matches()
+    }
+
+    fun modifyVisibility(
+        userName:EditText,
+        email: EditText,
+        number: EditText,
+        pincode: EditText,
+        address: EditText,
+        validate:Button,
+        cancel:Button,
+        confirm:Button,
+        visibility: Boolean
+    ){
+        userName.isEnabled=visibility
+        email.isEnabled=visibility
+        pincode.isEnabled=visibility
+        number.isEnabled=visibility
+        address.isEnabled=visibility
+        validate.visibility = if(!visibility) View.GONE else View.VISIBLE
+        cancel.visibility = if(visibility) View.GONE else View.VISIBLE
+        confirm.visibility =if(visibility) View.GONE else View.VISIBLE
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putBoolean("visibility",visibility)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
+        super.onRestoreInstanceState(savedInstanceState)
+        val v=savedInstanceState.getBoolean("visibility")
+        val userName = findViewById<EditText>(R.id.userNameEt)
```

Comment:
We do not have to initialise every time we need a view. Make a single instance and use it. Because findViewById takes some to find the view